dice-where-downloader-lib/src/main/java/technology/dice/dicewhere/downloader/destination/s3/S3FileAcceptor.java

```diff
@@ -58,22 +60,29 @@ public StreamConsumer<Void> getStreamConsumer(
                   .storageClass(StorageClass.INTELLIGENT_TIERING)
                   .build();
           client.putObject(putObjectRequest, RequestBody.fromInputStream(stream, size));
-
           Latest latest = new Latest(clock.instant(), key);
           String latestContent = mapper.writeValueAsString(latest);
 
-          PutObjectRequest putLatest =
-              PutObjectRequest.builder()
-                  .key(Paths.get(key).getParent().toString() + "/" + LATEST_KEY)
-                  .bucket(bucket)
-                  .contentLength((long) latestContent.length())
-                  .storageClass(StorageClass.INTELLIGENT_TIERING)
-                  .build();
-          client.putObject(
-              putLatest,
-              RequestBody.fromInputStream(
-                  new StringInputStream(latestContent), latestContent.length()));
+          if ((!noMd5Check) && (!originalFileMd5.matches(stream.md5()))) {
+            LOG.error("MD5 mismatch. Deleting destination file");
+            client.deleteObject(DeleteObjectRequest.builder()
```

Comment:
does this mean that the "consumer", in our use case dce-id has RW access to the S3 bucket?

Reply:
it should not. This code path is ran only by the fargate task.